abi3audit/_extract.py

```diff
@@ -222,10 +226,24 @@ def __init__(self, spec: PyPISpec) -> None:
         )
 
     def __iter__(self) -> Iterator[_object.SharedObject]:
+        # PyPI specs look like <PKG><OP><VERSION>,
+        # where <PKG> is any valid package name,
+        # <OP> is exactly one of ~=, ==, !=, <=, >=, <, >, ===,
+        # and <VERSION> is a package version as specified in
+        # https://packaging.python.org/en/latest/specifications/version-specifiers.
+        # TODO (nicholasjng): Allow version ranges, i.e. comma-separated
+        #  <OP><VERSION> pairs, like numpy>=1.3.0, <2.
+        match = re.fullmatch(_FULL_PYPI_RE, self.spec, re.VERBOSE | re.IGNORECASE)
```

Comment:
I need to parse here again, since the `PyPISpec` does not contain info about its constituent parts. `Version` and `SpecifierSet` also use regexes to parse the inputs their constructors (and fail if they are ill-formed), so this is not optimal at all in terms of performance.

I could relax the regex in `make_specs` above and let the aforementioned constructors throw the errors (which would also improve the error messages in the CLI), but since you're a security guy, I didn't want to make you angry and ask for permission first :=)

Also, the PyPI response has checksums for each wheel that we could verify before writing it to the wheelhouse, I'm already doing that in the tests.